client/http-client/src/transport.rs

```diff
@@ -100,11 +103,16 @@ impl HttpTransportClient {
 			return Err(Error::RequestTooLarge);
 		}
 
-		let req = hyper::Request::post(&self.target)
+		let mut req = hyper::Request::post(&self.target)
```

Comment:
this is not clone right? 

otherwise we could just build the request once and clone it on every call.

however can we pre-compute the "headers" at just set them with `req.headers_mut = self.headers.clone()` on each request?

seems annoying to that on every call.